File: win-prog/ProgWinCS/Font Fun/SimpleShear/SimpleShear.cs
//------------------------------------------
// SimpleShear.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class SimpleShear: FontMenuForm
{
     public new static void Main()
     {
          Application.Run(new SimpleShear());
     }
     public SimpleShear()
     {
          Text = "Simple Shear";

          strText = "Shear";
          font = new Font("Times New Roman", 72);
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Brush  brush = new SolidBrush(clr);
          Matrix matx  = new Matrix();

          matx.Shear(0.5f, 0);
          grfx.Transform = matx;

          grfx.DrawString(strText, font, brush, 0, 0);
     }
}

File: win-prog/ProgWinVB/Font Fun/SimpleShear/SimpleShear.vb
'--------------------------------------------
' SimpleShear.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class SimpleShear
    Inherits FontMenuForm

    Shared Shadows Sub Main()
        Application.Run(New SimpleShear())
    End Sub

    Sub New()
        Text = "Simple Shear"
        strText = "Shear"
        fnt = New Font("Times New Roman", 72)
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim br As New SolidBrush(clr)
        Dim matx As New Matrix()

        matx.Shear(0.5, 0)
        grfx.Transform = matx
        grfx.DrawString(strText, fnt, br, 0, 0)
    End Sub
End Class
